clients/unshaded/src/main/java/tachyon/client/FileSystemMasterClient.java

```diff
@@ -272,21 +272,21 @@ public synchronized void completeFile(long fileId) throws IOException, TachyonEx
   }
 
   /**
-   * Deletes a file.
+   * Deletes a file or a directory.
    *
-   * @param fileId the file id
+   * @param id the id
    * @param recursive whether to delete the file recursively (when it is a directory)
    * @return whether operation succeeded or not
    * @throws IOException if an I/O error occurs
    * @throws TachyonException if a Tachyon error occurs
    */
-  public synchronized boolean deleteFile(long fileId, boolean recursive) throws IOException,
+  public synchronized boolean delete(long id, boolean recursive) throws IOException,
       TachyonException {
     int retry = 0;
     while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
       connect();
       try {
-        return mClient.deleteFile(fileId, recursive);
+        return mClient.remove(id, recursive);
```

Comment:
see the notes `NOTE: Unfortunately, the method cannot be called "delete" as that is a reserved Thrift keyword.`